ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Client.scala

```diff
@@ -146,7 +146,7 @@ private[ember] class H2Client[F[_]: Async](
             case Some("http/1.1") => Resource.pure[F, SocketType](Http1)
             case Some(other) =>
               Resource.raiseError[F, SocketType, Throwable](
-                new Throwable("Unknown Protocol Received")
+                new ProtocolException(s"Unknown protocol")
```

Comment:
Alright, this can't happen. I just checked. Since we are the client we set the input alpn set, so it can only be one of those two in the output alpn set according to spec. 

`None` is a possibility, but this should never happen with a well-behaved server.